### ISSUE TITLE
BatchNorm1d, running_mean and running_var

### ISSUE THREAD

**Issue by FredSRichardson** [pytorch/pytorch#117217]
### 🚀 The feature, motivation and pitch

This is regarding the `running_mean` and `running_var` arrays associated with a `BatchNorm1d` layer.  These two vectors are not included in the `model.named_paramers()` list of tuples, but I think they should be.  The reason is that the values in these arrays significantly impact the output of `model.forward()` when just running inference.  For example, NaNs in those two arrays will make the output of the `BatchNorm1d` layer also NaNs if `track_running_stats` is True.  If `running_mean` and `running_var` were included in the `model.named_parameter()` output, this type of problem could easily be detected.

### Alternatives

Any method call that would go through all parameters used in the `forward()` method of a model and confirm that there are no anomalous values at inference time (*not* at train time - I believe `torch.autograd.set_detect_anomaly(True)` takes care of that) would take care of this for me.  Apologies if one already exists!  Google was not giving me anything helpful, but my search terms may not have been great.

### Additional context

I loaded a model that has NaNs in the `running_mean` and `running_var` arrays associated with a `BatchNorm1d` layer.   Iterating over `model.named_paramers()` lead me to the erroneous conclusion that there were no NaNs or Infs in my model (or any particular large or small values).  But I was getting NaNs in the output of `model.forward()` with apparently reasonable input values.  I had to step pretty carefully through the code to figure out that this was occurring in a `BatchNorm1d` call in the first layer.  Now I know the model is the root problem of this situation, what I'm really asking for is an easy way to detect bad (NaN, Inf, >+/-1E+19, etc) values in a model.  Or maybe there is one, in which case I apologize profusely!

**Comment by colesbury** [comment]
They are not in `named_parameters()` because they are not `parameters()` PyTorch's sense of the word. They are not learned via SGD (or whatever optimizer you use for the rest of your NN module), and `parameters()` are specifically those things passed to the optimizer. 

The `running_mean`/`var` show up in the module's `buffers()` and `state_dict()`.
